streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -121,6 +123,13 @@ public KStreamImpl(KStreamBuilder topology, String name, Set<String> sourceNodes
     }
 
     @Override
+    public <K1> KStream<K1, V> selectKey(KeyValueMapper<K, V, KeyValue<K1, V>> mapper) {
+        String name = topology.newName(KEY_SELECT_NAME);
+        topology.addProcessor(name, new KStreamMap<>(mapper), this.name);
```

Comment:
With the above change, this would be ```new KStreamMap<>(new KeyValueMapper<K, V, KeyValue<K1, V>>) { apply(k, v -> new KeyValue(mapper.apply(k, v), v))}```